### PR TITLE
Add reference doc for IPAMConfiguration

### PR DESCRIPTION
**Pull request by caseydavenport**
## Related issues/PRs

Documents https://github.com/projectcalico/calico/pull/6393

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: calico/reference/resources/ipamconfig.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+title: IPAM configuration
+description: IP address management global configuration
+canonical_url: '/reference/resources/ipamconfig'
+---
+
+A IPAM configuration resource (`IPAMConfiguration`) represents IPAM specific global configuration options.
```
Comment (lmm) [inline calico/reference/resources/ipamconfig.md:7]: super nit: `represents global IPAM configuration options` or `represents IPAM-specific global configuration options`

**Review by lmm — APPROVED** [review]
LGTM, tiny nit that can be ignored.

File: calico/reference/resources/ipamconfig.md (hunk 1 above)
Comment (coutinhop) [inline calico/reference/resources/ipamconfig.md:7]: English is not my first language so I could be way off here, but shouldn't it be "**An** IPAM configuration resource..."?

Comment (caseydavenport) [reply calico/reference/resources/ipamconfig.md]: Yep, definitely should be.

**Pull request merged**